packages/react/src/utils.ts

```diff
@@ -482,6 +482,312 @@ export const sortByProperty = <T>(arr: T[], property: keyof T, order: SortOrder
   });
 };
 
+export const formatShortDateString = (date: Date) => {
+  if (!date) return date;
+
+  const year = date.getFullYear();
+  const month = String(date.getMonth() + 1).padStart(2, "0");
+  const day = String(date.getDate()).padStart(2, "0");
+
+  return `${year}-${month}-${day}`;
+};
+
+export const formatLongDateString = (date: Date) => {
+  if (!date) return date;
+
+  return date.toLocaleString("en-US", { year: "numeric", month: "short", day: "numeric" });
+};
+
+export const formatLongDateTimeString = (date: Date) => {
+  if (!date) return date;
+
+  return date
+    .toLocaleString("en-US", {
+      year: "numeric",
+      month: "short",
+      day: "numeric",
+      hour: "numeric",
+      minute: "2-digit",
+      hour12: true,
+    })
+    .replace(/,(?=[^,]*$)/, "");
+};
+
+export const isValidDate = (date: unknown) => {
+  return date instanceof Date && !isNaN(date.getTime());
+};
+
+export const utcToZonedTime = (date: Date, timeZone: string): Date => {
+  const offsetMilliseconds = tzParseTimezone(timeZone, date, true);
+
+  const d = new Date(date.getTime() - offsetMilliseconds);
+
+  const resultDate = new Date(0);
+
+  resultDate.setFullYear(d.getUTCFullYear(), d.getUTCMonth(), d.getUTCDate());
+
+  resultDate.setHours(d.getUTCHours(), d.getUTCMinutes(), d.getUTCSeconds(), d.getUTCMilliseconds());
+
+  return resultDate;
+};
+
+const patterns = {
+  timezone: /([Z+-].*)$/,
+  timezoneZ: /^(Z)$/,
+  timezoneHH: /^([+-]\d{2})$/,
+  timezoneHHMM: /^([+-])(\d{2}):?(\d{2})$/,
+};
+
+const validIANATimezoneCache: Record<string, boolean> = {};
+
+const MILLISECONDS_IN_HOUR = 3600000;
+const MILLISECONDS_IN_MINUTE = 60000;
+
+const isValidTimezoneIANAString = (timeZoneString: string) => {
+  if (validIANATimezoneCache[timeZoneString]) return true;
+  try {
+    new Intl.DateTimeFormat(undefined, { timeZone: timeZoneString });
+    validIANATimezoneCache[timeZoneString] = true;
+    return true;
+  } catch (error) {
+    return false;
+  }
+};
+
+export const tzParseTimezone = (timezoneString: string | undefined, date: Date | number | undefined, isUtcDate?: boolean): number => {
```

Comment:
is this from `date-fns` library? if so we should credit that in the comments

Reply:
Good point: will update the PR